pkg/kubelet/pleg/generic.go

```diff
@@ -423,12 +424,16 @@ func getContainerState(pod *kubecontainer.Pod, cid *kubecontainer.ContainerID) p
 }
 
 func updateRunningPodAndContainerMetrics(pods []*kubecontainer.Pod) {
-	// Set the number of running pods in the parameter
-	metrics.RunningPodCount.Set(float64(len(pods)))
 	// intermediate map to store the count of each "container_state"
 	containerStateCount := make(map[string]int)
+	runningPods := 0
 
 	for _, pod := range pods {
+		// Count only Running pods
+		p := pod.ToAPIPod()
```

Comment:
Will this method `ToAPIPod ` update the `Status`? I cannot find where it does it:

https://github.com/kubernetes/kubernetes/blob/5ed7b1afb8958fe0d5ddd3660582add89ab9a372/pkg/kubelet/container/runtime.go#L598-L613

Reply:
I was hoping it did, but apparently it doesn't which can be seen in failing bazel-test. Any ideas how to get pod status/phase without iterating over all containers and thus duplicating logic?